clients/src/main/java/org/apache/kafka/common/protocol/Protocol.java

```diff
@@ -116,18 +118,20 @@ public static String toHtml() {
         final StringBuilder b = new StringBuilder();
         b.append("<h5>Headers:</h5>\n");
 
-        b.append("<pre>");
-        b.append("Request Header => ");
-        schemaToBnfHtml(RequestHeader.SCHEMA, b, 2);
-        b.append("</pre>\n");
-        schemaToFieldTableHtml(RequestHeader.SCHEMA, b);
-
-        b.append("<pre>");
-        b.append("Response Header => ");
-        schemaToBnfHtml(ResponseHeader.SCHEMA, b, 2);
-        b.append("</pre>\n");
-        schemaToFieldTableHtml(ResponseHeader.SCHEMA, b);
-
+        for (int i = 0; i < RequestHeaderData.SCHEMAS.length; i++) {
+            b.append("<pre>");
+            b.append("Request Header v").append(i).append(" => ");
+            schemaToBnfHtml(RequestHeaderData.SCHEMAS[i], b, 2);
+            b.append("</pre>\n");
+            schemaToFieldTableHtml(RequestHeaderData.SCHEMAS[i], b);
+        }
+        for (int i = 0; i < ResponseHeaderData.SCHEMAS.length; i++) {
+            b.append("<pre>");
+            b.append("Response Header v").append(i).append(" => ");
+            schemaToBnfHtml(ResponseHeaderData.SCHEMAS[i], b, 2);
+            b.append("</pre>\n");
+            schemaToFieldTableHtml(ResponseHeaderData.SCHEMAS[i], b);
+        }
```

Comment:
I want to execute this code locally. What is the `gradle` command to generate this?

Reply:
```
./gradlew genProtocolMessageDocs
```
Look for `protocol_messages.html`